pkg/sql/plan/shuffle.go

```diff
@@ -35,6 +37,37 @@ const (
 	ShuffleToMultiMatchedReg int32 = 2
 )
 
+// convert first 8 bytes to uint64
+func ByteSliceToUint64(bytes []byte) uint64 {
+	var result uint64 = 0
+	i := 0
+	length := len(bytes)
+	for ; i < 8; i++ {
+		result = result * 256
+		if i < length {
+			result += uint64(bytes[i])
+		}
+	}
+	return result
+}
+
+// convert first 8 bytes to uint64. vec.area must be nil
+// if varlena length less than 8 bytes, should have filled zero in varlena
+func VarlenaToUint64Inline(v *types.Varlena) uint64 {
+	return bits.ReverseBytes64(*(*uint64)(unsafe.Add(unsafe.Pointer(&v[0]), 1)))
+}
+
+// convert first 8 bytes to uint64
+func VarlenaToUint64(v *types.Varlena, area []byte) uint64 {
+	svlen := (*v)[0]
+	if svlen <= types.VarlenaInlineSize {
+		return VarlenaToUint64Inline(v)
+	} else {
+		voff, vlen := v.OffsetLen()
+		return ByteSliceToUint64(area[voff : voff+vlen])
```

Comment:
vlen must be bigger than 8, so you should still use the ReverseBytes version here

Reply:
fixed